src/broadcast/filelistener/filelistener.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include <QTextCodec>
+#include <QtConcurrentRun>
+
+#include "broadcast/filelistener/filelistener.h"
+#include "broadcast/filelistener/metadatafileworker.h"
+#include "preferences/metadatafilesettings.h"
+
+
+FileListener::FileListener(UserSettingsPointer pConfig)
+        : m_COsettingsChanged(kFileSettingsChanged),
+          m_pConfig(pConfig),
+          m_latestSettings(MetadataFileSettings::getPersistedSettings(pConfig)) {
+
+    MetadataFileWorker *newWorker = new MetadataFileWorker(m_latestSettings.filePath);
+    newWorker->moveToThread(&m_workerThread);
+
+    connect(&m_workerThread,SIGNAL(finished()),
+            newWorker,SLOT(deleteLater()));
+
+    connect(this,SIGNAL(deleteFile()),
+            newWorker,SLOT(slotDeleteFile()));
+
+    connect(this,SIGNAL(moveFile(QString)),
+            newWorker,SLOT(slotMoveFile(QString)));
+
+    connect(this,SIGNAL(writeMetadataToFile(QByteArray)),
+            newWorker,SLOT(slotWriteMetadataToFile(QByteArray)));
+
+    connect(this,SIGNAL(clearFile()),
+            newWorker,SLOT(slotClearFile()));
+
+    connect(&m_COsettingsChanged,SIGNAL(valueChanged(double)),
+            this,SLOT(slotFileSettingsChanged(double)));
+
+    m_workerThread.start();
+}
+
+FileListener::~FileListener() {
+    m_workerThread.quit();
+    m_workerThread.wait();
+}
+
+
+void FileListener::slotBroadcastCurrentTrack(TrackPointer pTrack) {
+    if (!pTrack)
```

Comment:
Can this really happen? If not, we should use VERIFY_AND_DEBUG_ASSERT.

Reply:
Yes, this happens every time a track is paused. The signal emitted by PlayerInfo passes a null pointer when the track is paused.

Reply:
Is the AllTracksPaused signal redundant?
IMHO the current track can be either replaced, or paused. But when the current track is paused, we should clear the file, right?

Reply:
Yes, perhaps you're right... the advantage of that slot is that it is immediate. But perhaps it doesn't matter. I don't know...